srv/salt/_modules/dg.py

```diff
@@ -815,6 +822,24 @@ def _reduce_inventory(self, disk: dict) -> dict:
                 raise
 
 
+def _apply_policies(wal_devices: list, db_devices: list) -> bool:
+    """ Apply known policies """
+    if wal_devices and not db_devices:
+        log.error("""
+        You specified only wal_devices. If your intention was to
+        have dedicated WALs/DBs please specify it with the db_devices
+        filter. WALs will be colocated alongside the DBs.
+        """)
+        return False
+
+    if wal_devices and db_devices:
+        if len(wal_devices) < len(db_devices):
+            log.error("This doesn't work right now.")
```

Comment:
Can we elaborate slightly? Something like "There are less wal devices than db devices"?

Reply:
Sure; The reason for this is described here: https://bugzilla.suse.com/show_bug.cgi?id=1135179

Quote:

```
Due to a design flaw in ceph-volume (http://tracker.ceph.com/issues/37502) we have to make sure that each ceph-volume call only receives *one* db_device as parameter in DeepSea.

In situations with three distinct wal / db/ data devices we have a couple of restrictions.

Examples:


data_devices = [/dev/sda, /dev/sdb, /dev/sdc, /dev/sdd, /dev/sde, ...]

db_devices = [/dev/sdw, /dev/sdx, /dev/sdy, /dev/sdz]

wal_devices = [/dev/sdo, /dev/sdp, /dev/sdq]



in this case would have to distribute the wal_devices unevenly across the db_devices.

With the current implementation we generate calls that look like this:

ceph-volume .. --db-devices /dev/sdw --wal-devices /dev/sdo 
ceph-volume .. --db-devices /dev/sdx --wal-devices /dev/sdp 
ceph-volume .. --db-devices /dev/sdy --wal-devices /dev/sdq 
ceph-volume .. --db-devices /dev/sdz <--- we can't guarantee a separate wal_device for this batch of OSDs.



This type of problem is seen when:

1) db_devices is 0 (was taken into account already) and wal_devices not 0
2) db_devices > wal_devices



Problem 2) can be tackled in multiple ways:


a) Fill(or stretch) wal_devices so that the last call would look like this:

ceph-volume .. --db-devices /dev/sdz --wal-devices /dev/sdo <-- (the first again)


b) Restrict this kind of configuration. Add policy for this case.
   Technically this is a bad thing to do as it adds a rather large failure domain.
```